uaclient/config.py

```diff
@@ -665,6 +700,37 @@ def help(self, name):
         response_dict["help"] = help_ent.help_info
         return response_dict
 
+    def write_cfg(self, config_path=None):
+        """Write config values back to config_path or DEFAULT_CONFIG_FILE."""
+        if not config_path:
+            config_path = DEFAULT_CONFIG_FILE
+        content = "# Ubuntu-Advantage client config file.\n"
+
+        for attr in ("contract_url", "security_url"):
+            content += "{key}: '{value}'\n".format(
+                key=attr, value=getattr(self, attr)
+            )
+        content += "data_dir: {}\n".format(self.data_dir)
+        if "log_level" in self.cfg:
+            log_level = self.cfg["log_level"].lower()
+        else:
+            log_level = CONFIG_DEFAULTS["log_level"]
+        content += "log_level: {}\n".format(log_level)
+        content += "log_file: {}\n".format(self.log_file)
+        ua_config = {
+            "ua_config": {
+                "http_proxy": self.http_proxy,
+                "https_proxy": self.https_proxy,
+                "apt_http_proxy": self.apt_http_proxy,
+                "apt_https_proxy": self.apt_https_proxy,
+            }
+        }
+        content += "\n# Managed with `sudo ua config`\n" + yaml.dump(ua_config)
+        for attr in ("features", "settings_overrides"):
+            if self.cfg.get(attr):
+                content += "\n" + yaml.dump({attr: self.cfg.get(attr)})
```

Comment:
Do these two need to be handled with separate `yaml.dump` calls? Can we construct a dict and call `yaml.dump` on the whole thing?

Note that we can use `default_flow_style=False` when we have nested dicts: https://pyyaml.org/wiki/PyYAMLDocumentation#dictionaries-without-nested-collections-are-not-dumped-correctly

Reply:
Switched to use default_flow_style=False here. Thanks for the tip.